admin/add_copyright.py

```diff
@@ -118,7 +121,21 @@ def main():
             if handler is not None:
                 with open(fullname) as f:
                     lines = list(line.rstrip() for line in f)
-                newLines = handler(fullname, copy.copy(lines))  # use a copy because of the compare in the next line
+                if filename == "__main__.py":
+                    copyright_string = ["COPYRIGHT = ("]
+                    for line in COPYRIGHT:
+                        copyright_string.append(f'   "{line}\\n"')
+                    copyright_string.append(")")
+                    copyright_string.append("")
+                    lines = re.sub(
+                        r"COPYRIGHT = \(\n(?:[^\n]+\n)+\)\n",
+                        "\n".join(copyright_string).replace("\\", "\\\\"),
+                        "\n".join(lines),
+                    ).split("\n")
+                    lines = [line.rstrip() for line in lines]
+                newLines = handler(
+                    fullname, copy.copy(lines)
+                )  # use a copy because of the compare in the next line
                 if newLines != lines:
```

Comment:
Modifying `lines` means that the `if newLines != lines` check will not work. Instead, the modification should be performed in a separate function that is then selected as a `handler` above.

Reply:
You're right. Another solution is to move this block into the `if newLines != lines:` block because if the copiright in the code only needs to be updated if the comment is updated.